pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java

```diff
@@ -65,13 +68,24 @@ public static void setExecutionStatistics(IntermediateResultsBlock resultsBlock,
       if (executionStatistics.getNumDocsScanned() > 0) {
         numSegmentsMatched++;
       }
+      try {
```

Comment:
Can you remove the try-catch block?

Reply:
@snleee I am not sure if we can remove that for queries with limit 0 like `SELECT * FROM testTable LIMIT 0` we are passing EmptySelectionOperator Instance in there which does not implement the getIndexSegment() hence there will be an exception we have a explicit test case to prevent that `SelectionCombineOperatorTest.testSelectionLimit0` Since we are using the interface (Operator) to reference in the for loop and interface has explicit check to throw the exception I think it is better to enclose it within the try catch block.

Also is the regression failing because of this try catch ? I could not see any logs associated with the same ? Can you please help me understand the failure so that I can fix the issue. Thanks

Reply:
@ravishankar15 I thought `getIndexSegment()` would return null but the interface actually throw `UnsupportedOperationException` and you're handling it correctly. 

Let me take a look on the regression test and try to check why it fails.